api/Dockerfile.dev

```diff
@@ -21,13 +21,16 @@ COPY requirements.txt $API_HOME
 RUN pip install --upgrade pip && pip install -r requirements.txt
 
 # copy api directory to docker's work directory. 
-COPY . $API_HOME  
+COPY . $API_HOME
 
 # make migrations
-RUN chmod +x scripts/makemigrations.sh && ./scripts/makemigrations.sh && python manage.py migrate
+RUN chmod +x scripts/makemigrations.sh && ./scripts/makemigrations.sh
+
+# Migrate all migrations
+RUN python manage.py migrate
 
 # port where the Django app runs  
-EXPOSE 8000  
+EXPOSE 8000
 
 # start server  
-ENTRYPOINT ["python", "manage.py", "runserver", "0.0.0.0:8000"]
+CMD ["python", "manage.py", "runserver", "0.0.0.0:8000"]
```

Comment:
Why CMD over ENTRYPOINT?

Reply:
Because ENTRYPOINT is the command that will always run when container starts and we can't overwrite it.
The simple case can be when someone don't want to run default command and want to start app opening bash first.
If someone doesn't overwrite the CMD command that will work as ENTRYPOINT.
Let me know if it is okay or we should migrate to ENTRYPOINT again.

Reply:
Hmm, goal of this `Dockerfile.dev` is to allow users to `run` the api or ui server. I can't think of a reason why we should allow overriding of this command.

If there is a case where it will be helpful to the developer then we should use CMD. For now, I can't think of any.

Reply:
As I mentioned above the case might be if we want to start server on our own using `bash`.
We can use ENTRYPOINT in prod Dockerfile.

Reply:
According to [Docker docs](https://docs.docker.com/engine/reference/builder/#understand-how-cmd-and-entrypoint-interact):
> ENTRYPOINT should be defined when using the container as an executable.

In our case, we are using it as an executable to run API or UI server.
Also, we can override the entrypoint using `--entrypoint` flag.

Reply:
My view is the ENTRYPOINT is more suitable for production and for development setup we should use CMD because as I debug containers I don't use CMD or ENTRYPOINT and open bash to start the server.
If this is not the case than I will change CMD to ENTRYPOINT again.

Reply:
Cool, I am okay with using CMD in dev env. Make sure to use ENTRYPOINT for prod.

Reply:
Sure, Thank you !!